src/main/java/org/duckstudy/racer/Racer.java

```diff
@@ -0,0 +1,72 @@
+package org.duckstudy.racer;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import org.duckstudy.movingcar.Car;
+import org.duckstudy.movingcar.CarMover;
+
+public class Racer {
+
+    private static final int INTIAL_POSITION = 0;
+    private final CarMover carMover;
+    private final ArrayList<Car> carList = new ArrayList<>();
+    private final int repetitionNum;
+
+    public Racer(CarMover carMover, int participantNum, int repetitionNum, String[] nameList)
+            throws IllegalArgumentException {
+        if (validateNameList(nameList)) {
+            throw new IllegalArgumentException("자동차 이름은 1글자 이상 5자 이하만 가능합니다.");
+        }
+        if (validateRepetitionNum(repetitionNum)) {
+            throw new IllegalArgumentException("반복 횟수는 0보다 커야 합니다.");
+        }
+
+        this.carMover = carMover;
+        addCar(participantNum, nameList);
+        this.repetitionNum = repetitionNum;
+    }
+
+    private boolean validateNameList(String[] nameList) {
+        return Arrays.stream(nameList).anyMatch(name -> name.isEmpty() || name.length() > 5);
```

Comment:
```suggestion
        return Arrays.stream(nameList)
            .anyMatch(name -> name.isEmpty() || name.length() > 5);
```
가 조금 더 가독성이 높을 것 같아요.

Reply:
.을 줄마다 작성해야하는데 놓친 것 같습니다 감사합니당 !!